SimTracker/TrackAssociation/plugins/TrackTimeValueMapProducer.cc

```diff
@@ -48,27 +53,25 @@ class TrackTimeValueMapProducer : public edm::global::EDProducer<> {
 private:
   // inputs
   const edm::EDGetTokenT<edm::View<reco::Track> > tracks_;
-  const edm::EDGetTokenT<edm::View<reco::Track> > gsfTracks_;
+  const std::string tracksName_;
   const edm::EDGetTokenT<TrackingParticleCollection> trackingParticles_;
   const edm::EDGetTokenT<TrackingVertexCollection> trackingVertices_;
-  const edm::EDGetTokenT<edm::HepMCProduct> hepMCProduct_;
+  const edm::EDGetTokenT<std::vector<PileupSummaryInfo> > pileupSummaryInfo_;
   // tracking particle associators by order of preference
   const std::vector<edm::EDGetTokenT<reco::TrackToTrackingParticleAssociator> > associators_;  
+  // eta bounds
+  const float etaMin_, etaMax_, ptMin_, pMin_, etaMaxForPtThreshold_;
   // options
   std::vector<std::unique_ptr<const ResolutionModel> > resolutions_;
   // functions
-  void calculateTrackTimes( const edm::View<reco::Track>&, 
-                            const std::vector<reco::RecoToSimCollection>&,
-                            std::vector<float>& ) const;
-  std::pair<float,float> extractTrackVertexTime(const std::vector<std::pair<TrackingParticleRef, double> >&) const;
+  float extractTrackVertexTime(const TrackingParticle&, const reco::TransientTrack&) const;
 };
 
 DEFINE_FWK_MODULE(TrackTimeValueMapProducer);
 
 namespace {
-  constexpr float fakeBeamSpotTimeWidth = 0.175f; // ns
-  const std::string generalTracksName("generalTracks");
-  const std::string gsfTracksName("gsfTracks");
+  constexpr float fakeBeamSpotTimeWidth = 0.300f; // ns
```

Comment:
shouldn't this be configurable or come from the common beam spot smearing parameters?
What is the equivalent value now in actual simulation?

Reply:
This is in fact not used anymore (the number is computed from the rms of the pileup vertex times from the pileupsummaryinfo)

Reply:
I mean that the magic number is not used anymore, and the value is computed on the fly from the pileupsummaryinfo, such that this line can be removed in later cleanup.

Reply:
can we drop the constant definition then if it's not needed/used?